src/validation/validateSubmissions.ts

```diff
@@ -0,0 +1,256 @@
+        validateSearchResults.hasErrors = true
+        validateSearchResults.errors?.push({
+            field: 'orderBy',
+            errorCode: ErrorCode.ORDERBY_OPTION_INVALID,
+            httpStatus: 400
+        })
+    }
+
+    const invalidOrderByOption = filters.orderBy?.find(order => 
+        !validOrderByFields.includes(order.fieldToOrder))
+
+    if (invalidOrderByOption) {
+        validateSearchResults.hasErrors = true
+        validateSearchResults.errors?.push({
+            field: 'orderBy',
+            errorCode: ErrorCode.ORDERBY_OPTION_NOT_FOUND,
+            httpStatus: 400
+        })
+    }
+
+    if (filters.orderBy && filters.orderBy.length > 2) {
+        validateSearchResults.hasErrors = true
+        validateSearchResults.errors?.push({
+            field: 'orderBy',
+            errorCode: ErrorCode.ORDERBY_OPTION_MAX_LIMIT,
+            httpStatus: 400
+        })
+    }
+
+    if (filters.orderBy && filters.orderBy.length === 2 && 
+        filters.orderBy[0].fieldToOrder === filters.orderBy[1].fieldToOrder) {
+        validateSearchResults.hasErrors = true
+        validateSearchResults.errors?.push({
+            field: 'orderBy',
+            errorCode: ErrorCode.ORDERBY_FIELD_SELECTED_TWICE,
+            httpStatus: 400
+        })
+    }
+}
+
+function validateSubmissionSearchBySpokenLanguage(
+    filters: gqlTypes.SubmissionSearchFilters,
+    validateSearchResults: Result<unknown>
+    ): void {
+
+    if (!filters.spokenLanguages) {
+        return
+    }
+
+    if (filters.spokenLanguages.length == 0) {
+        validateSearchResults.hasErrors = true
+        validateSearchResults.errors?.push({
+            field: 'spokenLanguages',
+            errorCode: ErrorCode.MISSING_INPUT,
+            httpStatus: 400
+        })
+    }
+    
+    for (const lang of filters.spokenLanguages) {
+        if (!Object.values(gqlTypes.Locale).includes(lang)) {
+            validateSearchResults.hasErrors = true
+            validateSearchResults.errors?.push({
+                field: 'spokenLanguages',
+                errorCode: ErrorCode.INVALID_LANGUAGE_INPUT,
+                httpStatus: 400
+            })
+        }
+    }
+    
+    const languageCodeOccurrences: any = {}
+    for (let lang of filters.spokenLanguages) {
+        if(languageCodeOccurrences[lang]) {     
+            validateSearchResults.hasErrors = true
+            validateSearchResults.errors?.push({
+                field: 'spokenLanguages',
+                errorCode: ErrorCode.INVALID_LANGUAGE_INPUT,
+                httpStatus: 400
+            })
+            break
+        } else {
+            languageCodeOccurrences[lang] = 1
+        }
+    }
+}
+
+function validateSearchByName(
+    filters: gqlTypes.SubmissionSearchFilters,
+    validateSearchResults: Result<unknown>
+): void {
+    const name = filters.healthcareProfessionalName
+
+    if (name === undefined || name === null) {
```

Comment:
can simplify this to 
```suggestion
    if (!name) {
```

Reply:
@ermish I've made this change. But it now allows empty strings (`''`) as search inputs. Is this okay?